src/include/kdbmacros.h

```diff
@@ -73,21 +82,30 @@
 			Key * c = ksNext (info);                                                                                           \
 			if (!c)                                                                                                            \
 			{                                                                                                                  \
-				ELEKTRA_SET_ERRORF (84, error, "the key %s (value %s) was added", keyName (k), keyString (k));             \
+				/* TODO: Correct? Maybe Interface? */                                                                      \
+				ELEKTRA_SET_INSTALLATION_ERRORF (                                                                          \
+					error, "Read only plugin, kdbSet not supported but the key %s (value %s) was added", keyName (k),  \
+					keyString (k));                                                                                    \
 				ksDel (info);                                                                                              \
 				return -1;                                                                                                 \
 			}                                                                                                                  \
 			if (strcmp (keyName (k), keyName (c)) || strcmp (keyString (k), keyString (c)))                                    \
 			{                                                                                                                  \
-				ELEKTRA_SET_ERRORF (84, error, "the key %s (expected %s) was modified to %s (expected %s)", keyName (k),   \
-						    keyName (c), keyString (k), keyString (c));                                            \
+				/* TODO: Correct? Maybe Interface? */                                                                      \
+				ELEKTRA_SET_INSTALLATION_ERRORF (                                                                          \
```

Comment:
I think it could be a validation error (read-only keys). Definitely not installation or interface.

Reply:
Validation error does not fit in this context. Installation might not fit because I assume that a different installation or compilation of elektra will suddenly make the plugin also support writing of keys. But since interface errors indicate a `misuse` as written in the guideline I think it actually fits here because someone misuses the plugin for setting keys. `Compared to semantic validation errors, this category
has its focus on detecting wrong usages of the API`. I think using a plugin in a not intended way fits into this category

Reply:
No, these read-only parts are always read-only, it is not dependent on compilation or installation. It only depends on the specification, like other validation.